R/get.R

```diff
@@ -0,0 +1,41 @@
+#' Retrieves participants present in secuTrialdata
+#'
+#' Given a secuTrialdata object, this function simply returns a list of participants.
+#' Information included are participant IDs and corresponding study centre information,
+#' if available.
+#'
+#' @param dat secuTrialdata object containing participant IDs and centre information
+#' @return data frame containing a list of participants present in dat
+#' @export
+#'
+#' @examples
+#'
+#' path <- system.file("extdata", "sT_exports", "export_options",
+#'                     "s_export_CSV-xls_CTU05_20191003-144349_all_info.zip",
+#'                     package = "secuTrialR")
+#' sT_export <- read_secuTrial(path)
+#'
+#' # show participants
+#' participants <- get_participants(sT_export)
+#'
+get_participants <- function(dat){
+
+  meta <- unlist(dat$export_options$meta_names)
+
+  if (dat$export_options$add_id & dat$export_options$centre_info){
+    participants <- dat[[meta["casenodes"]]][, c("mnppid", "mnpaid", "mnpctrid")]
+    centres <- dat[[meta["centres"]]][, c("mnpctrid", "mnpctrname")]
+    participants <- suppressMessages(left_join(participants, centres))
+  } else if (!dat$export_options$add_id & dat$export_options$centre_info){
+    participants <- dat[[meta["casenodes"]]][, c("mnppid", "mnpctrid")]
+    centres <- dat[[meta["centres"]]][, c("mnpctrid", "mnpctrname")]
+    participants <- suppressMessages(left_join(participants, centres))
+  } else if (dat$export_options$add_id & !dat$export_options$centre_info){
+    participants <- dat[[meta["casenodes"]]][, c("mnppid", "mnpaid")]
+  }
+  else{
+    participants <- dat[[meta["casenodes"]]][, "mnppid", drop = FALSE]
+  }
```

Comment:
this is painful to read due to the if elses... can it not be done more simply (get casenodes and then merge things into it or remove things from it)?

Reply:
thanks for your input, alan. i agree it's not so nice to read. 

the setup of the casenode table columns depends a lot on the export options. the presence of the centre table depends on export options too. so my impression is that if i don't select the correct columns within a series of "if elses", i would have to catch the exceptions generated by trying to fetch columns that might not exist due to selected export options. this might be even harder to read. 

what do you think? do you see another implementation route?

Reply:
is this enough...?
```
  participants <- dat[[meta["casenodes"]]]
  if(dat$export_options$centre_info){
    centres <- dat[[meta["centres"]]]
    participants <- suppressMessages(left_join(participants, centres))
  }
  participants <- participants[, which(names(participants) %in% c("mnppid", "mnpaid", "mnpctrid", "mnpctrname"))]
```
so, just merge in centers if it's there, and then retain variables if they're there....

Reply:
might need to check if the final result is a vector or dataframe though... could happen if only one of the four variables is available...

Reply:
Cool, thanks a lot for the clever suggestion. I'll give it a try now.

Reply:
did it work? much easier to read huh? 😃

Reply:
Yeah it worked and I like how it looks :) I added drop = FALSE to extract it as a data.frame on all occasions.

Reply:
perfect